### PR TITLE
Add configs for Foam Cutter by rcKeith

### PR DESCRIPTION
**Pull request by DerAndere1**
### Requirements

4 axis hot wire foam cutter by rcKeith with RAMPS board (https://rckeith.co.uk/4-axis-cnc-foam-cutter-using-marlin-3d-printer-firmware/ )

### Description

This adds example configs for a 4 axis hot wire foam cutter CNC machine which is well documented in the blog post by rcKeith at https://rckeith.co.uk/4-axis-cnc-foam-cutter-using-marlin-3d-printer-firmware/ .

The modified pins_RAMPS.h file is also provided. Maybe those modifications have to be documented in the README instead?

EDIT: The modifications of pin_RAMPS.h are now done in the Marlin repo ( PR https://github.com/MarlinFirmware/Marlin/pull/24325)

## Benefits

Better example configs for a hot wire foam cutter using the `FOAMCUTTER_XYUV` feature

### Related Issues

https://github.com/MarlinFirmware/Configurations/issues/750
https://github.com/MarlinFirmware/Marlin/pull/24325

### CODE REVIEW

File: config/examples/FoamCutter/rcKeith/Foam_Cutter/Configuration.h [hunk 1]
```diff
@@ -0,0 +1,3100 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+#pragma once
+
```
Comment (thisiskeithb) [inline config/examples/FoamCutter/rcKeith/Foam_Cutter/Configuration.h]: You're missing the "Don't build with.." errors needed for all `import-2.1.x` configs:
https://github.com/MarlinFirmware/Configurations/blob/c27da0bf6a3ff090851a350cc2a897668b4dc85c/config/default/Configuration.h#L23-L24

**Comment by thisiskeithb** [comment]
> The modified pins_RAMPS.h file is also provided. Maybe those modifications have to be documented in the README instead?

It'd be better to include it in the main Marlin repo, perhaps with some kind of config flag or as a new board.

**Comment by DerAndere1** [comment]
Thx for the review. Technically this is not a new board. Conditionally changing pin definitions in pins_RAMPS in Marlin depending on `FOAMCUTTER_XYUV` feature flag sounds like a solution. This PR should be treated as a draft until this is sorted

File: config/examples/FoamCutter/rcKeith/Foam_Cutter/Configuration.h (hunk 1 above)
Comment (DerAndere1) [reply config/examples/FoamCutter/rcKeith/Foam_Cutter/Configuration.h]: fixed

**Pull request merged**